src/sentry/static/sentry/app/utils/analytics.tsx

```diff
@@ -196,3 +220,29 @@ metric.measure = function metricMeasure({name, start, end, data = {}, noCleanup}
     metricDataStore.delete(start);
   }
 };
+
+/**
+ * Used to pass data between startTransaction and endTransaction
+ */
+const transactionDataStore = new Map<string, object>();
+
+const getCurrentTransaction = () => {
+  return Sentry.getCurrentHub()
+    .getScope()
+    ?.getTransaction();
+};
+
+metric.startTransaction = ({name, traceId, op}) => {
+  if (!traceId) {
+    traceId = getCurrentTransaction()?.traceId;
```

Comment:
Nice, does this show up in the parent transaction then?

Reply:
doesn't show up in the parent transaction as a span, but searching by trace id shows all the related transactions. Perhaps I should be setting parent_span_id to correctly link them

Reply:
I think being able to see the relation with traceid for now is probably enough. We'll want to turn these back into spans later I think, once we've figured out how to properly wrap our app and avoid idleTimeout killing the txn too early